### PR TITLE
Mobile 3833

### PR DESCRIPTION
**Pull request by crazyserver**

### CODE REVIEW

File: src/addons/mod/glossary/pages/edit/edit.html [hunk 1]
```diff
@@ -11,76 +11,73 @@ <h1 *ngIf="glossary">
...
+                </ion-label>
+            </ion-item-divider>
+            <core-attachments [files]="attachments" [component]="component" [componentId]="glossary.coursemodule" [allowOffline]="true"
+                [courseId]="courseId">
+            </core-attachments>
+            <ng-container *ngIf="glossary.usedynalink">
                 <ion-item-divider>
                     <ion-label>
-                        <h2>{{ 'addon.mod_glossary.attachment' | translate }}</h2>
+                        <h2>{{ 'addon.mod_glossary.linking' | translate }}</h2>
```
Comment (NoelDeMartin) [inline src/addons/mod/glossary/pages/edit/edit.html:61]: Did this really change? (and if so, why?) Or is it just some diffing defect? I assume you only changed the component for the directive here and the other html templates.

File: src/core/classes/page-transition.ts [hunk 2]
```diff
@@ -229,11 +229,18 @@ export const moodleTransitionAnimation = (navEl: HTMLElement, opts: TransitionOp
 
         rootAnimation.addAnimation(leavingContent);
 
+        const hasTransform = !!leavingContentEl && (leavingContentEl as HTMLElement).style.transform !== '';
```
Comment (NoelDeMartin) [inline src/core/classes/page-transition.ts:232]: I'm not sure I understand what this is doing, other than explaining maybe you could add an inline comment or refactor the code to make it more readable? (encapsulating code in other functions, etc.)

**Review by NoelDeMartin — CHANGES_REQUESTED** [review]
I found a couple of things to improve, but looks good overall.

File: src/addons/mod/glossary/pages/edit/edit.html (hunk 1 above)
Comment (crazyserver) [reply src/addons/mod/glossary/pages/edit/edit.html:61]: it's a defect... :-p

File: src/core/classes/page-transition.ts (hunk 2 above)
Comment (crazyserver) [reply src/core/classes/page-transition.ts:232]: I think it's ok now

File: src/core/directives/swipe-navigation.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        return CoreScreen.isMobile && !!this.manager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async ngAfterViewInit(): Promise<void> {
+
+        const hasNext = await this.manager?.hasNextItem();
+        const hasPrevious = await this.manager?.hasPreviousItem();
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/directives/swipe-navigation.ts:53]: great, remove and moved after the threshold check

File: src/core/directives/swipe-navigation.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            canStart: () => this.enabled,
+            onStart: () => {
+                style.transition = '';
+            },
+            onMove: (ev) => {
+                style.transform = `translateX(${ev.deltaX * 0.6 }px)`;
+            },
+            onEnd: (ev) => {
+                style.transition = '.5s ease-out';
+                if (hasNext && ev.deltaX > 150) {
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/directives/swipe-navigation.ts:74]: done

File: src/core/directives/swipe-navigation.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+// limitations under the License.
+
+import {  AfterViewInit, Directive, ElementRef, Input, OnDestroy } from '@angular/core';
+import { CoreSwipeNavigationItemsManager } from '@classes/items-management/swipe-navigation-items-manager';
+import { Gesture } from '@ionic/angular';
+import { CoreScreen } from '@services/screen';
+import { GestureController } from '@singletons';
+
+/**
+ * Directive to enabled content navigation with swipe.
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/directives/swipe-navigation.ts:22]: Yep, done.

**Review by NoelDeMartin — APPROVED** [review]

**Pull request merged**